### PR TITLE
Ignore does not work with integers #112

### PR DESCRIPTION
**Pull request by yoyonel**
PR for this issue: [Ignore does not work with integers #112](https://github.com/inveniosoftware/dictdiffer/issues/112)

```
(py3_dictdiffer) ➜  dictdiffer git:(master) ./run-tests.sh
Skipped 1 files
lists of files in version control and sdist match
running pytest
running egg_info
writing dictdiffer.egg-info/PKG-INFO
writing dependency_links to dictdiffer.egg-info/dependency_links.txt
writing requirements to dictdiffer.egg-info/requires.txt
writing top-level names to dictdiffer.egg-info/top_level.txt
reading manifest file 'dictdiffer.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'dictdiffer.egg-info/SOURCES.txt'
running build_ext
======================================================================= test session starts =======================================================================
platform linux -- Python 3.6.8, pytest-4.2.0, py-1.7.0, pluggy-0.8.1
rootdir: /home/.../dictdiffer, inifile: pytest.ini
plugins: pep8-1.0.6, cov-2.6.1
collected 101 items

tests/test_conflict.py .....                                                                                                                                [  4%]
tests/test_dictdiffer.py ...................................................                                                                                [ 55%]
tests/test_merge.py ....                                                                                                                                    [ 59%]
tests/test_resolve.py .........                                                                                                                             [ 68%]
tests/test_unify.py ...                                                                                                                                     [ 71%]
tests/test_utils.py ........                                                                                                                                [ 79%]
dictdiffer/__init__.py ....                                                                                                                                 [ 83%]
dictdiffer/_compat.py .                                                                                                                                     [ 84%]
dictdiffer/conflict.py .                                                                                                                                    [ 85%]
dictdiffer/merge.py ..                                                                                                                                      [ 87%]
dictdiffer/resolve.py ..                                                                                                                                    [ 89%]
dictdiffer/unify.py .                                                                                                                                       [ 90%]
dictdiffer/utils.py .........                                                                                                                               [ 99%]
dictdiffer/version.py .                                                                                                                                     [100%]

----------- coverage: platform linux, python 3.6.8-final-0 -----------
Name                     Stmts   Miss  Cover   Missing
------------------------------------------------------
dictdiffer/__init__.py     130      0   100%
dictdiffer/_compat.py       17      4    76%   14-15, 23-24
dictdiffer/conflict.py      28      0   100%
dictdiffer/merge.py         42      0   100%
dictdiffer/resolve.py       51      0   100%
dictdiffer/unify.py         19      0   100%
dictdiffer/utils.py        107      0   100%
dictdiffer/version.py        2      0   100%
------------------------------------------------------
TOTAL                      396      4    99%


=================================================================== 101 passed in 1.23 seconds ====================================================================
```

### CODE REVIEW

File: dictdiffer/__init__.py [hunk 1]
```diff
@@ -25,9 +25,9 @@
 
 __all__ = ('diff', 'patch', 'swap', 'revert', 'dot_lookup', '__version__')
 
-DICT_TYPES = (MutableMapping, )
-LIST_TYPES = (MutableSequence, )
-SET_TYPES = (MutableSet, )
+DICT_TYPES = (MutableMapping,)
```
Comment (jirikuncar) [inline dictdiffer/__init__.py]: @yoyonel can you please move the whitespace changes to separate PR with an automatic style check (e.g. YAPF)?

File: dictdiffer/__init__.py [hunk 2]
```diff
@@ -124,7 +124,13 @@ def diff(first, second, node=None, ignore=None, path_limit=None, expand=False,
             for value in ignore
         }
 
-    node = node or []
+    if ignore:
```
Comment (jirikuncar) [inline dictdiffer/__init__.py]: can we merge the code with the previous condition `if isinstance(ignore, list):`?

Something around these lines:
```
if ignore:
    def _process_ignore_value(value):
        # TODO ...

    ignore = type(ignore)(_process_ignore_value(value) for value in ignore)
```

File: dictdiffer/__init__.py [hunk 3]
```diff
@@ -134,126 +140,133 @@ def dotted(node, default_type=list):
...
-                    else:
-                        collect.append((key, second[key].__class__()))
-                        recurred = diff(second[key].__class__(),
-                                        second[key],
-                                        node=node+[key],
-                                        ignore=ignore,
-                                        path_limit=path_limit,
-                                        expand=expand,
-                                        tolerance=tolerance)
+    def _diff_recurrence(_first, _second, _node=None):
```
Comment (jirikuncar) [inline dictdiffer/__init__.py]: maybe a better name would be `_diff_recursive` since it is called it from its body.

File: setup.py [hunk 4]
```diff
@@ -32,21 +32,15 @@
...
-    'docs': [
-        'Sphinx>=1.4.4',
-        'sphinx-rtd-theme>=0.1.9',
-    ],
-    'numpy': [
-        'numpy>=1.11.0',
-    ],
-    'tests': tests_require,
-}
+extras_require = {':python_version=="2.6"': [
```
Comment (jirikuncar) [inline setup.py]: please keep the indentation style

File: tests/test_dictdiffer.py [hunk 5]
```diff
@@ -208,8 +208,8 @@ def test_remove_list(self):
         assert ('remove', 'a', [(1, 'c'), (0, 'b'), ]) == diffed
 
     def test_add_set(self):
-        first = {'a': set([1, 2, 3])}
-        second = {'a': set([0, 1, 2, 3])}
+        first = {'a': {1, 2, 3}}
```
Comment (jirikuncar) [inline tests/test_dictdiffer.py]: we can change it only after officially removing Python 2.6 support.

File: dictdiffer/__init__.py (hunk 1 above)
Comment (yoyonel) [reply dictdiffer/__init__.py]: Yep, no pb :)

File: dictdiffer/__init__.py (hunk 3 above)
Comment (yoyonel) [reply dictdiffer/__init__.py]: If u want, no pb for me

File: tests/test_dictdiffer.py (hunk 5 above)
Comment (yoyonel) [reply tests/test_dictdiffer.py]: oki my bad

Comment (jirikuncar) [reply tests/test_dictdiffer.py]: Sorry, you can keep it. We have removed the 2.6 some time ago. I have rechecked `setup.py`.

File: dictdiffer/__init__.py (hunk 2 above)
Comment (yoyonel) [reply dictdiffer/__init__.py]: oki done :)

**Comment by yoyonel** [comment]
You're welcome :)
